src/security_utilities_rust/src/microsoft_security_utilities_core/marvin.rs

```diff
@@ -2,23 +2,24 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 /// This is a Rust implementation of the Marvin32 checksum algorithm, the definitive native code for which is
-/// at https://github.com/microsoft/SymCrypt/blob/master/lib/marvin32.c. 
-use std::mem;
+/// at https://github.com/microsoft/SymCrypt/blob/master/lib/marvin32.c.
 
 /// Convenience method to compute a Marvin hash and collapse it into a 32-bit hash.
-pub fn compute_hash32(data: &[u8], seed: u64, offset: i32, length: i32) -> i32
-{
-    let hash64: i64 = compute_hash(data, seed, offset, length);
+pub fn compute_hash32(data: &[u8], seed: u64, offset: i32, length: i32) -> i32 {
+    let hash64 = compute_hash(data, seed, offset, length);
     return ((hash64 >> 32) as i32) ^ (hash64 as i32);
 }
 
-/// Computes a 64-bit hash using the Marvin algorithm.
-pub fn compute_hash(data: &[u8], seed: u64, offset: i32, mut length: i32) -> i64
-{
-    // Marvin by design can produce a checksum for empty input buffers, which
-    // is why it's ok for the offset to point just past the end of the buffer
-    // or for the input buffer to be empty;
-    if offset < 0 || offset > data.len() as i32
+/// Convenience method to compute a Marvin hash from a slice and collapse it into a 32-bit hash.
+pub fn compute_hash32_slice(data: &[u8], seed: u64) -> i32 {
+    let hash64 = compute_hash_slice(data, seed);
+    return ((hash64 >> 32) as i32) ^ (hash64 as i32);
+}
+
+/// Computes a 64-bit hash using the Marvin algorithm from the given slice,
+/// using the provided length and offset to determine the data to hash.
+pub fn compute_hash(data: &[u8], seed: u64, offset: i32, length: i32) -> i64 {
+    if offset > data.len() as i32
```

Comment:
>offset > data.len() as i32 [](http://example.com/codeflow?start=7&length=26)

The `offset < 0` was removed: this is a public function. What prevents someone from sending a negative offset?

Reply:
Excellent catch, I missed that. Fixed